spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/sampler/UuidToBigIntegerConverterTest.java

```diff
@@ -0,0 +1,52 @@
+package org.springframework.cloud.sleuth.sampler;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import lombok.Data;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.math.BigInteger;
+import java.util.UUID;
+
+import static java.math.BigInteger.ONE;
+import static java.math.BigInteger.ZERO;
+import static junitparams.JUnitParamsRunner.$;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.springframework.cloud.sleuth.sampler.UuidToBigIntegerConverterTest.TestData.sample;
+
+@RunWith(JUnitParamsRunner.class)
+public class UuidToBigIntegerConverterTest {
+
+	private static final BigInteger TWO = new BigInteger("2");
+
+	@Test
+	@Parameters
+	public void should_convert_uuid_to_big_integer(TestData testData) throws Exception {
+		UUID uuid = new UUID(testData.mostSignificantBits, testData.leastSignificantBits);
+
+		BigInteger bigInteger = UuidToBigIntegerConverter.uuidToBigInt(uuid);
+
+		then(bigInteger).isEqualByComparingTo(testData.expectedInteger);
+	}
+
+	public Object[] parametersForShould_convert_uuid_to_big_integer() {
+		return $(
+				sample(Long.MAX_VALUE, Long.MAX_VALUE, TWO.pow(126).subtract(ONE)),
```

Comment:
As the Long's MAX_VALUE javadoc says

```
/**
     * A constant holding the maximum value a {@code long} can
     * have, 2<sup>63</sup>-1.
     */
    @Native public static final long MAX_VALUE = 0x7fffffffffffffffL;
```

So we get

1...63x1...64x1

the first 1 is a sign bit.
the 127 bits are the remaining bits thus the greatest value possible is
2^127-1.

Per analogy if we have 3 bits then the greatest is 111 which is 8.

We have 128 bits where there is also a bit sign. So 127 bits. And in the
Long's Max value we have the additional -1. Alltogether 2^127 - 1.

I also tried your approach but first it should be:

new BigInteger(
  ByteBuffer.allocate(16).putLong(Long.MAX_VALUE).putLong(Long.MAX_VALUE).array()
)

Not 128 cause it's in bytes not bits, but still it's not working as it
should ;)